### DISCUSSION TITLE
Configuration for Playwright's experimental components testing

### DISCUSSION THREAD

**Discussion by phcoliveira** [vercel/next.js#41190] [Ideas]
### Describe the feature you'd like to request

I would like to use a recommended configuration for testing a React component, with Next.JS features such as `next/link`, with Playwright's experimental components testing.

Currently, there is a recommended configuration for Jest using JS DOM as a testing environment. Such configuration is very handy because it allows to mount a React component in isolation, independently of having a Next.JS page.

Playwright's experimental components testing can already mount a React component. But the usage of Next.JS features in isolation breaks the tests. So let us please provide such boilerplate.

### Describe the solution you'd like

I would like to have a file, analagous to `packages/next/build/jest/jest.ts`, in order to provide the necessary boilerplate for testing React components with Next.JS features in isolation.

Then, inside a custom file named `playwright.config.{js|ts}`, at the root of a Next.JS project, I could do something like:

```TypeScript
import nextPlaywrightCT from 'next/playwright-ct'

const createNextPlaywrightCT = nextPlaywrightCT({ dir: './' })

export default createNextPlaywrightCT({ ... })
```

The snippet above mimicks the usage of `nextJest`, which returns a function usually named as `createNextJest`. I do not find strictly necessary to have such function. It could also be an object to be "spread" inside a regular `PlaywrightTestConfig` object.

Alternatively, it could be simpler to have a fixture named `nextMount` that would take care of loading the required boilerplate. Playwright's fixtures are awesome, so this could be a better way.

Something like:

```TypeScript
import {test as base, expect} from '@playwright/experimental-ct-react'
import {fixtures, type TestFixtures, type WorkerFixtures} from 'next/playwright-ct'

const test = base.extend<TestFixtures, WorkerFixtures>({
  ...fixtures // such as `nextMount`
})

test('it mounts a component with Next.JS features', async ({nextMount}) => {
  const component = nextMount(<MyComponentWithNextLink />)

  expect(component).toBeVisible()
```

There is still room for discussing about how such a link could be clicked and tested.

### Describe alternatives you've considered

I have been looking for a way to develop a `global-setup` file for Playwright in order to mimick the boilerplate required to test a component in isolation. Playwright CT has mainly two files: `index.html` and `index.ts`, for loading anything needed to setup the testing environment.

Inside the file `index.ts` I have tried to [load the testing environment variables](https://nextjs.org/docs/basic-features/environment-variables#test-environment-variables) as described in documentation.

I have also tried to mount a component inside a fake `RouterContext.Provider`, [as described here](https://github.com/vercel/next.js/issues/15543).

So far, I got no success.

**Comment by mikestopcontinues** [comment]
Did you ever get this working?

**Comment by sre-is-toby** [comment]
+1 on if this works as well :)

**Comment by phcoliveira** [comment]
> Did you ever get this working?

No. I am sorry for the late response. But I gave up on Next Js as soon as I found that such a thing is also not possible with jsDom.
You have to move some mountains with next-router-mock.
